django/contrib/admin/templates/admin/includes/fieldset.html

```diff
@@ -1,4 +1,5 @@
-{% with name=fieldset.name|default:""|slugify %}
+{% load admin_modify %}
+{% with name=fieldset.name|default:""|slugify_allow_unicode %}
 <fieldset class="module aligned {{ fieldset.classes }}"{% if name %} aria-labelledby="{{ prefix }}-{{ id_prefix}}-{{ name }}-{{ id_suffix }}-heading"{% endif %}>
```

Comment:
As far as I'm aware, all Unicode characters are fine for the `ID` attribute. CSS identifiers are more limited, but also allow Unicode characters, and we don't have any CSS rules based on these IDs.

Reply:
As an alternative, I can remove `slugify_allow_unicode` and do:
```diff
--- a/django/contrib/admin/templates/admin/includes/fieldset.html
+++ b/django/contrib/admin/templates/admin/includes/fieldset.html
@@ -1,7 +1,6 @@
-{% load admin_modify %}
-{% with name=fieldset.name|default:""|slugify_allow_unicode %}
-<fieldset class="module aligned {{ fieldset.classes }}"{% if name %} aria-labelledby="{{ prefix }}-{{ id_prefix}}-{{ name }}-{{ id_suffix }}-heading"{% endif %}>
-    {% if name %}
+{% with name=fieldset.name|default:""|slugify %}
+<fieldset class="module aligned {{ fieldset.classes }}"{% if fieldset.name %} aria-labelledby="{{ prefix }}-{{ id_prefix}}-{{ name }}-{{ id_suffix }}-heading"{% endif %}>
+    {% if fieldset.name %}
         {% if fieldset.is_collapsible %}<details><summary>{% endif %}
         <h{{ heading_level|default:2 }} id="{{ prefix }}-{{ id_prefix}}-{{ name }}-{{ id_suffix }}-heading" class="fieldset-heading">{{ fieldset.name }}</h{{ heading_level|default:2 }}>
         {% if fieldset.is_collapsible %}</summary>{% endif %}
diff --git a/tests/admin_views/tests.py b/tests/admin_views/tests.py
index ffcde0c0e5..7d9f2c7031 100644
--- a/tests/admin_views/tests.py
+++ b/tests/admin_views/tests.py
@@ -2545,7 +2545,7 @@ class AdminViewPermissionsTest(TestCase):
         )
         self.assertContains(
             response,
-            '<h2 id="fieldset-0-이름-2-heading" class="fieldset-heading">이름</h2>',
+            '<h2 id="fieldset-0--2-heading" class="fieldset-heading">이름</h2>',
         )
```
I think this would also work?

Reply:
:thinking: I'd totally remove `fieldset.name` from the `ID`, if we don't need it. I see no reason to keep it, TBH :shrug:

Reply:
I am not opposed to removing the name entirely, but it feels like an out-of-scope and potentially breaking change, which is not desirable in a release blocker.

Reply:
> I am not opposed to removing the name entirely, but it feels like an out-of-scope and potentially breaking change, which is not desirable in a release blocker.

I have a gut feeling that we will get more complains in the future, if we will keep it. Moreover, the pattern used for `ID`'s is an undocumented implementation detail, not a part of public API, so our API stability policy doesn't apply here.

Reply:
> Moreover, the pattern used for ID's is an undocumented implementation detail, not a part of public API, so our API stability policy doesn't apply here.

Fair. Let's get rid of it. 🥳